sdk/core/azure-core/azure/core/pipeline/transport/_base.py

```diff
@@ -68,7 +69,7 @@
 HTTPResponseType = TypeVar("HTTPResponseType")
 HTTPRequestType = TypeVar("HTTPRequestType")
 PipelineType = TypeVar("PipelineType")
-DataType = Optional[Union[bytes, Dict[str, Union[str, int]]]]
+DataType = Optional[Union[bytes, str, Dict[str, Union[str, int]]]]
```

Comment:
minor but we can be explicit when `DataType` can be null

```suggestion
DataType = Union[bytes, str, Dict[str, Union[str, int]]]
```